tests/bdp_util.erl

```diff
@@ -0,0 +1,221 @@
+%% -------------------------------------------------------------------
+%%
+%% Copyright (c) 2015 Basho Technologies, Inc.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+%% @doc Helper functions for bdp_service_manager.erl and others.
+
+-module(bdp_util).
+
+-export([build_cluster/1, build_cluster/2,
+         service_added/4, service_removed/2, service_started/4, service_stopped/4,
+         make_node_leave/2, make_node_join/2]).
+-export([get_services/1, wait_services/2]).
+
+-include_lib("eunit/include/eunit.hrl").
+
+-type group_name() :: string().
+-type config_name() :: string().
+-type service_type() :: string().
+-type service_config() :: [{string(), string()}].
+-type service() :: {group_name(), config_name(), node()}.
+-type package() :: {config_name(), service_type(), service_config()}.
+
+-define(SM_RPC_RETRIES, 20).
+
+%% @ignore
+%% This is a workaround for the nasty habit of Service Manager to kick
+%% back with an {error, timeout} in response to any call (in
+%% particular, any rpc calls from riak_test over to dev1).  The error
+%% is returned immediately, which is weird, but goes away after some 8
+%% seconds of trying.  It's reasonable to stick that to SM itself and
+%% get it fixed there; for the time being, this workaround is here.
+
+call_with_patience(Node, M, F, A) ->
+    call_with_patience_(Node, M, F, A, ?SM_RPC_RETRIES).
+call_with_patience_(_Node, _M, _F, _A, 0) ->
+    lager:error("Exhausted ~b retries for an RPC call to ~p ~p:~p/~b",
+                [?SM_RPC_RETRIES, _Node, _M, _F, length(_A)]),
+    ?assert(false);
```

Comment:
Rather than just using an assert false that give no real indication of what when wrong, throwing a descriptive error will have a clearer failure:

     error({retries_exhausted, {Node, M, F, A}});